### ISSUE TITLE
[Bug] Unable to get length of string in v3.25+

### ISSUE THREAD

**Issue by jrjohnson** [emberjs/ember.js#19400]
### 🐞 Describe the Bug
Using `{{get this.someString "length"}}` on a string worked up through `v3.24.2`, but fails in `v3.25.0+`.

### 🔬 Minimal Reproduction
In any template (I did this in `application.hbs` of a fresh ember app).

```hbs
{{get this.someString "length"}}
{{log (get this.someString "length")}}
```

### 😕 Actual Behavior
Nothing displays in template and console reports `undefined`.

### 🤔 Expected Behavior
Should output the string length in the template and then in the console.

### 🌍 Environment

- Ember: 3.24.2
- Node.js/npm: v14.15.4
- OS: -OSX
- Browser: Firefox/Chrome/Safari

**Comment by pzuraq** [comment]
I think it's because of some differences in the implementation of the `{{get}}` helper after we upstreamed it, mainly: https://github.com/emberjs/ember.js/blob/75bb0c42dcf526c92ced96c404dc30edcc09c1da/packages/@ember/-internals/glimmer/lib/helpers/get.ts#L107-L113

This branch was removed. I believe this is a latent bug, because with a dynamic path on 3.24 and prior this wouldn't have worked either, it only worked because the path was the constant value `"length"`. Either way, we should fix this, you can `get` on non-objects clearly

**Comment by rwjblue** [comment]
Reopening until we get the rendering engine update landed here (and a 3.25.x release queued up).

**Comment by jrjohnson** [comment]
Thanks!
